cmake/defaults/Packages.cmake

```diff
@@ -152,8 +156,23 @@ endif()
 
 
 # --TBB
-find_package(TBB REQUIRED COMPONENTS tbb)
-add_definitions(${TBB_DEFINITIONS})
+if (PXR_ENABLE_ONEAPI_TBB)
+    find_package(TBB REQUIRED COMPONENTS tbb12) # OneAPI TBB encodes the major version on the library name.
```

Comment:
I think it would be preferable to put as much logic as possible in `FindTBB.cmake`, and to avoid having a `PXR_ENABLE_ONEAPI_TBB` option.

From what I can tell the library name can be `tbb12` or `tbb` on Windows, and is always `tbb` on other platforms.

In Embree it looks for both on Windows:
https://github.com/embree/embree/blob/master/common/cmake/FindTBB.cmake#L319

While in OpenImageIO it only looks for `tbb`:
https://github.com/OpenImageIO/oiio/blob/master/src/cmake/modules/FindTBB.cmake

Reply:
I updated #2466 now with a fix for `tbb12` detection on Windows. Other build system changes in this PR may be needed as well, but their purpose is not clear to me.